src/runtime_src/core/tools/common/ReportAsyncError.cpp

```diff
@@ -0,0 +1,108 @@
+namespace qr = xrt_core::query;
+
+boost::property_tree::ptree
+populate_async_error(const xrt_core::device * device)
+{
+  boost::property_tree::ptree pt;
+  boost::property_tree::ptree error_array;
+  auto dhdl = xrtDeviceOpenFromXcl(device->get_device_handle());
+  for (xrtErrorClass ecl = XRT_ERROR_CLASS_FIRST_ENTRY; ecl < XRT_ERROR_CLASS_UNKNOWN; ecl = xrtErrorClass(ecl+1)) {
+    xrtErrorCode errorCode = 0;
+    uint64_t timestamp = 0;
+    int rval = xrtErrorGetLast(dhdl, ecl, &errorCode, &timestamp);
+    /**
+     * In case of no error for given class, errorCode and timestamp will be zero,
+     * but zero errorCode is valid error, so checking for only timestamp.
+     */
+    if (rval == 0 && errorCode && timestamp) {
+      size_t len = 0;
+      if (xrtErrorGetJson(dhdl, errorCode, nullptr, 0, &len))
+        continue;
+      char buf[len];
+      if (xrtErrorGetJson(dhdl, errorCode, buf, len, nullptr))
+        continue;
+      std::stringstream ss(buf);
+      boost::property_tree::ptree _pt;
+      boost::property_tree::read_json(ss, _pt);
+      boost::property_tree::ptree node;
+      node.put("timestamp", timestamp);
+      node.put("class.code", _pt.get<int>("class.code"));
+      node.put("class.string", _pt.get<std::string>("class.string"));
+      node.put("module.code", _pt.get<int>("module.code"));
+      node.put("module.string", _pt.get<std::string>("module.string"));
+      node.put("severity.code", _pt.get<int>("severity.code"));
+      node.put("severity.string", _pt.get<std::string>("severity.string"));
+      node.put("driver.code", _pt.get<int>("driver.code"));
+      node.put("driver.string", _pt.get<std::string>("driver.string"));
+      node.put("number.code", _pt.get<int>("number.code"));
+      node.put("number.string", _pt.get<std::string>("number.string"));
+      error_array.push_back(std::make_pair("", node));
+    }
+  }
+  pt.add_child("errors", error_array);
+
+  return error_array;
+}
+
+void
+ReportAsyncError::getPropertyTreeInternal( const xrt_core::device * _pDevice, 
+                                              boost::property_tree::ptree &_pt) const
+{
+  getPropertyTree20202(_pDevice, _pt);
+}
+
+void 
+ReportAsyncError::getPropertyTree20202( const xrt_core::device * _pDevice, 
+                                           boost::property_tree::ptree &_pt) const
+{
+  _pt.add_child("asynchronous_errors", populate_async_error(_pDevice));
+}
+
+void 
+ReportAsyncError::writeReport( const xrt_core::device * _pDevice,
+                                  const std::vector<std::string> & /*_elementsFilter*/, 
+                                  std::iostream & _output) const
+{
+  boost::property_tree::ptree _pt;
+  boost::property_tree::ptree empty_ptree;
+  getPropertyTreeInternal(_pDevice, _pt);
+
+  _output << "Asynchronous Errors\n";
+  _output << boost::format("  %-35s%-20s%-20s%-20s%-20s%-20s\n") % "Time" % "Class" % "Module" % "Driver" % "Severity" % "Error Code";

```

Comment:
Please create a boost::format variable and use it for BOTH outputs below.  For example:
```
boost::format fmt("  %-35s%-20s%-20s%-20s%-20s%-20s\n");
_output << fmt % "Time" % "Class" % "Module" % "Driver" % "Severity" % "Error Code";
```

Reply:
done